api/cpp/include/slint_platform.h

```diff
@@ -268,6 +268,49 @@ struct PhysicalRegion
     PhysicalRegion(cbindgen_private::types::IntRect inner) : inner(inner) { }
 };
 
+/// A 16bit pixel that has 5 red bits, 6 green bits and 5 blue bits
+struct Rgb565Pixel
+{
+    /// The red component, encoded in 5 bits.
+    uint16_t r : 5;
+    /// The green component, encoded in 6 bits.
+    uint16_t g : 6;
+    /// The blue component, encoded in 5 bits.
+    uint16_t b : 5;
+
+    /// Default constructor.
+    constexpr Rgb565Pixel() : r(0), g(0), b(0) { }
+
+    /// \brief Constructor that constructs from an Rgb8Pixel.
+    constexpr Rgb565Pixel(const Rgb8Pixel &pixel)
```

Comment:
Shouldn't this be explicit perhaps, since it involves a "loss of precision"? (alternative: a `from_rgb8` function)

Reply:
I made it explicit.

To me, whether something should be explicit or not is a semantic question question of conversion.  Does it make sense to convert a rgb8 pixel into a rgb565 pixel?  If no, then it should be explicit.
(example: can an integer be converted to a `vector<string>`. That doesn't make sense to me, so it should be explicit)

I think it makes sense to convert rgb8 to rgb565, but i see the problem of the loss of conversion. So I made it explicit. (although the conversion from uint32 to uint16, for example, is implicit, but you could say it's a "bug" in the language)

Reply:
If a conversion from an integer to a `vector<string>` doesn't make sense (which I agree with), then why would there be a constructor on `vector<string>` that takes an `integer` in the first place?

To me the presence of a constructor implies that the conversion makes sense in the first place. When it's a no-brainer (minimal cost, no data loss), then I think it makes sense to be implicit. But if it requires the use of the brain (so to say), i.e. it requires the developer to explicitly _accept_ a compromise (of performance or data loss for example), then making it explicit enforces that mental step.

Reply:
Because the constructor is not only about conversion, but also about initialization.  And you can initialize a vector with an integer (that's why the constructor that takes an integer exist, to allocates so so many elements)

This is also why i think the constructor of SkiaRenderer should be explicit, because sementically, a tuple `{ NativeWindowHandle, PhysicalSize}`  is not really a SkiaRenderer. On the other hand, a tuple of to integer is a Point or a Size.


> To me the presence of a constructor implies that the conversion makes sense in the first place. When it's a no-brainer (minimal cost, no data loss), then I think it makes sense to be implicit

I think that's where our opinion diverge. A constructor implies that it can be initialized with these arguments, while only an implicit constructor implies that the conversion makes sense.

But I understand your point about having it as explicit also as a way to visualize the performance or data loss impact. Hence i made it explicit.